specification/security_model.adoc

```diff
@@ -1,9 +1,242 @@
+    is assigned to a TVM. By accessing the device MMIO space, the host component
+    can program direct memory access destination addresses to its own address
+    space. Device generated data intended to be copied to the TVM is accessed by
+    the host component instead.
+
+|===
+
+==== CoVE-IO-T002
+
+.CoVE-IO-T002
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Tamper
+| Privileged host software
+| In scope
+| Host component writes to TVM confidential memory
+
+5+| **Description** - A privileged host software component programs a TDI that
+    is assigned to a TVM. By accessing the device MMIO space, the host component
+    can modify the programmed direct memory access destination addresses, and
+    redirect it to unintended parts of the TVM address space.
+
+|===
+
+==== CoVE-IO-T003
+
+.CoVE-IO-T003
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Denial of service
+| Privileged host software
+| **Not** in scope
+| TVM can not access a TDI that is assigned to it
+
+5+| **Description** - A privileged host software component resets or powers down
+    an assigned TDI or its physical device, while the TDI is assigned to a TVM.
+    The TVM is no longer able to directly access its assigned TDI.
+
+|===
+
+==== CoVE-IO-T004
+
+.CoVE-IO-T004
+|===
+
+| Asset | Threat | Adversary | Scope | Result
+
+| Device trusted MMIO
+| Tamper
+| Privileged host software
+| In scope
+| TVM programs a TDI that is unassigned to it
+
+5+| **Description** - A privileged host software component remaps a TVM assigned
+    TDI MMIO guest physical address to an unassigned TDI MMIO host physical
+    address. The TVM programs a different TDI than the one that is assigned to
+    it.
+
+|===
+
+==== CoVE-IO-T005
+
+.CoVE-IO-T005
+|===
+
+| Asset | Threat | Adversary | Scope | Result
+
+| Device trusted MMIO
+| Denial of service
+| Privileged host software
+| In scope
+| TVM can not access the trusted MMIO space of a TDI that is assigned to it
+
+5+| **Description** - A privileged host software component configures PCIe
+    switches to redirect or drop MMIO accesses from the TVM to one of its
+    assigned TDIs.
+
+|===
+
+==== CoVE-IO-T006
+
+.CoVE-IO-T006
+|===
+
+| Asset | Threat | Adversary | Scope | Result
+
+| Device trusted MMIO
+| Tamper
+| Privileged host software
+| In scope
+| TVM accesses an unassigned TDI trusted MMIO space
+
+5+| **Description** - A privileged host software component maps a TDI trusted
+    MMIO space into TVM1 as part of the TDI assignement. Then it unassigns the
+    TDI from TVM1 and assigns it to TVM2, withouth unmapping the TDI trusted
+    MMIO space from TVM1. TVM1 can tamper with a TDI trusted MMIO while it is
+    not assigned to it.
+
+|===
+
+==== CoVE-IO-T007
+
+.CoVE-IO-T007
+|===
+
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Tamper and Disclosure
```

Comment:
I notice that T001 and T002 separate Tamper and Disclosure. Here they are combined in T007. 

Question: What is the criteria to separate or combine?

Reply:
Good point. I am not sure what is the preferred way, but I think we should combine? T001 and T002 can be achieved through the same vector. What do you think?

Reply:
Yes, I am OK to merge, if the attack mechanisms are very similar.

But if the attack mechanisms are different, then we should separate them.

Reply:
I merged 001 and 002 together.